src/ore/src/antichain.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Antichain utilities.
+
+#![allow(missing_debug_implementations)]
+
+use timely::order::PartialOrder;
+use timely::progress::change_batch::ChangeBatch;
+use timely::progress::frontier::MutableAntichain;
+
+/// A token that holds a mutable antichain and invokes an action should it change.
+pub struct AntichainToken<T: PartialOrder + Ord + Clone> {
+    antichain: MutableAntichain<T>,
+    // This is behind an `Rc` so that `Clone` can use `&self` rather than `&mut self`.
+    changes: std::rc::Rc<std::cell::RefCell<ChangeBatch<T>>>,
+    action: std::rc::Rc<std::cell::RefCell<dyn FnMut(std::vec::Drain<(T, i64)>)>>,
```

Comment:
is this in a Rc RefCell to make cloning easier? (that's what I ran into when I tried to replace it with `Box`?)

and I guess making it so that `Clone` can use `&self` applies for this too

Reply:
I think so? That comment was from a thing I copied from Frank but that's what I gather.